src/wallet/wallet.cpp

```diff
@@ -752,8 +752,13 @@ bool CWallet::IsSpent(const COutPoint& outpoint) const
         const uint256& wtxid = it->second;
         const auto mit = mapWallet.find(wtxid);
         if (mit != mapWallet.end()) {
-            int depth = GetTxDepthInMainChain(mit->second);
-            if (depth > 0  || (depth == 0 && !mit->second.isAbandoned()))
+            // An output is considered spent by a spending transaction
+            // unless the spending transaction is conflicted or
+            // abandoned. The check below is written conservatively to
+            // check for the inverse condition and only consider spending
+            // transactions in known, potentially active states.
+            const auto& wtx = mit->second;
+            if (wtx.isConfirmed() || wtx.InMempool() || (wtx.isInactive() && !wtx.isAbandoned()))
```

Comment:
In commit "wallet: use CWalletTx member functions to determine tx state" (92b3204b69e5ec25cbb29bd158f25d2b07e48cf6)

Would suggest simplifying this code and deleting the comment so this is just:

```c++
const auto& wtx = mit->second;
if (!wtx.isAbandoned() && !wtx.isBlockConflicted())
```

Using the simpler condition would make this commit easier to understand, and make the change in the next commit more obvious where the line becomes:

```c++
if (!wtx.isAbandoned() && !wtx.isMempoolConflicted() && !wtx.isBlockConflicted())
```